src/MacroExpander.js

```diff
@@ -249,6 +249,40 @@ export default class MacroExpander implements MacroContextInterface {
         throw new Error(); // eslint-disable-line no-unreachable
     }
 
+    /**
+     * Fully expand the given macro name and return the resulting list of
+     * tokens, or return `undefined` if no such macro is defined.
+     */
+    expandMacro(name: string): Token[] | void {
```

Comment:
Return type can be `?Token[]` (same for below)

Reply:
They are technically different: `?Token[]` includes `null` and `undefined`, while `Token[] | void` includes just `undefined`.  (I learned the `| void` trick looking elsewhere at this codebase.)  I agree the former is simpler looking, while the latter is more precise...  Do you still think I should change it? (Also happy to hear any other votes.)

Reply:
I think it's fine as is.  I find `?Token[]` confusing b/c it's hard to tell whether it means `?(Token[])` or `(?Token)[]`.